contracts/eden/src/actions/induct.cpp

```diff
@@ -81,12 +82,23 @@ namespace eden
 
       globals globals{get_self()};
       inductions inductions{get_self()};
-      accounts accounts{get_self()};
+      accounts user_accounts{get_self()};
+      accounts internal_accounts{get_self(), "owned"_n};
 
       const auto& induction = inductions.get_induction(id);
       eosio::check(payer == induction.invitee(), "only inductee may donate using this action");
       eosio::check(quantity == globals.get().minimum_donation, "incorrect donation");
-      accounts.sub_balance(payer, quantity);
+      user_accounts.sub_balance(payer, quantity);
+      migrations migrations{get_self()};
+      if (auto migration = migrations.get<migrate_account_v0>())
```

Comment:
It'd be nice to roll migration handling into add_balance and sub_balance, or create a higher level of add_balance and sub_balance

Reply:
The main difficulty is that each use case (donation, deposit, and withdrawal) behaves differently.

Reply:
Also, the set of actions that need to handle migration is fixed, as actions yet to be added can just assert that this migration is complete.